contracts/WhitelistedCommitment.sol

```diff
@@ -89,35 +87,42 @@ contract WhitelistedCommitment is AccessControlled, AccessRoles, PublicCommitmen
             curve.burnNeumark(neumarks);
         }
     }
-    event Debug(uint256 value);
-    /// overrides base class to compute neumark reward for fixed price investors
+
+    /// overrides base class to compute neumark reward for ordered whitelistfix investors
     function giveNeumarks(address investor, uint256 eth, uint256 euros)
         internal
         returns (uint256)
     {
-        uint256 fixedTicket = fixedCost[investor]; // returns 0 in case of investor not in mapping
-        Debug(fixedTicket);
-        Debug(eth);
+        uint256 fixedRemainingTicket = fixedCostTickets[investor]; // returns 0 in case of investor not in mapping
+        uint256 fixedRemainingNeumarks = fixedCostNeumarks[investor]; // returns 0 in case of investor not in mapping
+
         // what is above limit for fixed price should be rewarded from curve
         uint256 reward = 0;
-        if ( eth > fixedTicket ) {
-            if (fixedTicket > 0) // recompute euro if part of msg.value goes thru whitelist
-                euros = convertToEUR(eth - fixedTicket);
-            reward = curve.issue(euros); // PublicCommitment.giveNeumarks(investor, eth - fixedTicket, euros);
-            eth = fixedTicket;
+        if ( eth > fixedRemainingTicket ) {
+            if (fixedRemainingTicket > 0) // recompute euro if part of msg.value goes thru whitelist
+                euros = convertToEUR(eth - fixedRemainingTicket);
+            reward = curve.issue(euros);
+            eth = fixedRemainingTicket;
         }
-        Debug(eth);
+
         // get pro rata neumark reward for any eth left
         uint256 fixedreward = 0;
         if (eth > 0) {
-            fixedreward = proportion(totalFixedCostNeumarks, eth, totalFixedCostAmount);
-            // rounding errors, send out remainders
-            // @remco review
-            uint256 remainingBalance = neumarkToken.balanceOf(address(this));
-            if (absDiff(fixedreward, remainingBalance) < 1000)
-                fixedreward = remainingBalance; // send all
-            // decrease ticket size
-            fixedCost[investor] -= eth;
+            fixedreward = proportion(fixedRemainingNeumarks, eth, fixedRemainingTicket);
```

Comment:
`proportion` should not be called with the second argument larger than the third (we should not have fractions >1 there) because of overflows and stuff.

It's much more elegant here to take
```
uint256 ticketEth = min(eth, fixedRemainingTicket);
```
This prevents overflows and we can elliminate the complex conditional that follows.

Reply:
this is not necessary as first we spend the surplus eth with whitelist mechanism and when the proportion is computed eth <= fixedRemainingTicket. Surely some intermediate variable should be added that makes code more readable and this will be done